packages/api-server/api_server/models/tortoise_models/authorization.py

```diff
@@ -0,0 +1,31 @@
+from tortoise.fields import (
+    CharField,
+    ForeignKeyField,
+    ForeignKeyRelation,
+    ReverseRelation,
+)
+from tortoise.models import Model
+
+
+class Role(Model):
+    name = CharField(255, pk=True)
+    permissions: ReverseRelation["ResourcePermission"]
+
+
+class ResourcePermission(Model):
+    # "obj" in casbin speak
+    # This has no foreign key because resources can be given any arbitrary group, sometimes even dynamicly.
+    authz_grp = CharField(255, index=True)
+    # "sub" in casbin speak
+    role: ForeignKeyRelation[Role] = ForeignKeyField("models.Role")
+    action = CharField(255)
```

Comment:
With this design, the `authz_grp`and `action` can be any arbitrary string. `role` should not be nullable as it is a foreign key (actually not sure if tortoise-orm use nullable foreign key by default).

For `action`, we could use `CharEnumField`, but I thought a `CharField` is good enough and leave it more flexible.

For `authz_grp`, we can't use an enum because we don't necessarily know what authorization group is possible.

Reply:
Got it! Thanks for explaining =) Makes sense.